### DISCUSSION TITLE
Can I control the cursor?

### DISCUSSION THREAD

**Discussion by feedhopper** [chhoumann/quickadd#307] [Q&A]
I would like to add to a file of ideas. The file is getting longer. I wish to add a # to create a new header formatting and then simply write away, directly into this line.

The setup how I have it right now is making the cursor jump to the top of the document. Am I missing an option?

This is my setup for this quickadd:

![Screen-2022-11-07_09-52-21](https://user-images.githubusercontent.com/61922985/200268122-c1a86c48-b01c-4c01-9e09-c272b7a1f3ae.jpg)

**Comment by lullinglore** [comment]
You shouldn't have "Insert After" enabled. You already have "Write to bottom of the file" enabled, that should be enough for what you want.
If you want to insert your capture after a specific already existing line in the note, you should copy-paste it inside the "Insert after" field. It looks like it's empty right now, so that's most likely why it's not working as intended.

That should solve your problem!

**Comment by feedhopper** [comment]
Hi Lulullia, and thanks for getting back.

Unfortunatey, it does not solve my problem: I tried from the start the way you described. then added the "Insert after" option as it was not working how I wanted.
Switching backnow does again not have the desired result.

Any other ideas?

**Comment by feedhopper** [comment]
If I leave out the "Capture format", I at least get to input a line. This allows me to write something like "# This is my new idea" and it gets properly added to the bottom of the list.

Then, however, the cursor jumps right to the top.

**Reply by lullinglore** [reply]
Did you disable "Insert after" completely? Also keep in mine the line you want things inserted after must be copy-pasted exactly. I think it's also case-sensitive.

Also for your capture, I was about to suggest it but, you need to add {{VALUE}} to your Capture Format.
`# {{VALUE}}`

This will ask you to input something, then it'll get captured to the desired location.

**Reply by feedhopper** [reply]
Hmmm.

The {{VALUE}} option is good. It works nicely, now creating a headline based on my input.
Irritatingly, though: The cursor is still in line 1

Is this not the case for you?

I just disabled all my plugins using "Divide and Conquer"
https://github.com/chrisgrieser/obsidian-divide-and-conquer
But still, the cursor jumps right to the top after inserting my {{VALUE}}

**Reply by lullinglore** [reply]
I tried different things, it looks like you can't actually control the cursor with QuickAdd. It's probably because when you're capturing something you usually don't want to open the file.

What you can do is, if you have Templater plugin, add `<% tp.file.cursor(1) %>` right after {{VALUE}}. Then, when you use the capture, you setup a hotkey for the command "Templater: Jump to next cursor location" that will directly put your cursor at your capture's position. A bit complicated but I can't think of another way to do this.

I can suggest an alternative solution if you want but it'll involve changing your workflow a bit. Or you could try opening a feature request.

**Comment by feedhopper** [comment]
Ahh. I am not sure I am getting this. If I use this method, I need to start the capture and activate the hotkey afterwards, then the cursor jumps to the cursors last position?

And, just to understand the command: 
In the <% tp.file.cursor(1) %>: does the (1) mean "last position" before the actual one?

**Reply by lullinglore** [reply]
With this method, here's what you'll have to do:
- Run your QuickAdd capture.
- Input whatever you like in the capture, hit enter
- The file will open, but the cursor will be at line 1.
- You then run the command (or hotkey you set) "Templater: Jump to next cursor location"

Then your cursor will be where you want it to, aka the capture you just made.

Also the (1) is just needed for the command to work. You can have multiple of these commands in a note, and the number is used to determine which command to run next. But since in this case there's only one of such commands it doesn't really matter.

Actually. While writing this out I realized there's definitely a way to automate the process above, using QuickAdd macros:
![image](https://user-images.githubusercontent.com/47358820/201425669-0d661a6a-8ed7-4415-a6ed-d2332a2367c1.png)

This will make it so your cursor will be placed right after your capture (as long as you add that templater command after {{VALUE}}) automatically with no additional actions on your end.

If you need a more detailed help for setting this up, feel free to ask! (sorry I didn't think of this earlier; it would have saved us some time)

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

QuickAdd can do this natively now, no Templater needed. Two options depending on the flow you want:

1. If the goal is "insert where I am and keep typing": in the Capture's **Write position** dropdown (with _Capture to active file_ on), pick **At cursor** or **New line below cursor**. The heading lands right where you are, so the cursor never jumps.

2. For your exact setup - append `# {{VALUE}}` to the bottom of the ideas file, then write directly under it: make a Macro with two steps: your Capture (bottom of active file), then the editor command **Move cursor to file end**. Run the macro instead of the capture and the cursor follows the new heading automatically. Those cursor commands (file start/end, line start/end) were added in 2.12.0; the cursor-relative write positions in 2.10.0.

So the `tp.file.cursor` workaround discussed above is no longer needed.

Docs: https://quickadd.obsidian.guide/docs/Choices/CaptureChoice

Closing as resolved.
